core/src/mast/node/basic_block_node/arbitrary.rs

```diff
@@ -194,42 +198,214 @@ impl Default for MastForestParams {
     }
 }
 
-/// Strategy for generating MastForest instances
-pub fn mast_forest_strategy(params: MastForestParams) -> impl Strategy<Value = MastForest> {
-    // BasicBlockNode generation must reference decorator IDs in [0, decorators)
-    let bb_params = BasicBlockNodeParams {
-        max_decorator_id_u32: params.decorators,
-        ..Default::default()
-    };
+impl Arbitrary for MastForest {
+    type Parameters = MastForestParams;
+    type Strategy = BoxedStrategy<Self>;
 
-    // 1) Generate a Vec<BasicBlockNode> with length in `params.blocks`
-    prop::collection::vec(any_with::<BasicBlockNode>(bb_params), params.blocks.clone())
-        // 2) Map concrete blocks -> build a concrete MastForest
-        .prop_map(move |blocks| {
-            let mut forest = MastForest::new();
+    fn arbitrary_with(params: Self::Parameters) -> Self::Strategy {
+        // BasicBlockNode generation must reference decorator IDs in [0, decorators)
+        let bb_params = BasicBlockNodeParams {
+            max_decorator_id_u32: params.decorators,
+            ..Default::default()
```

Comment:
Not from this PR, but the default value for `BasicBlockNodeParams::max_ops_len` is `32`. We should probably make this value greater than the max number of ops in a batch (72) - maybe `72 * 2`? So that we can catch bugs that only surface when we have multiple batches.

Reply:
So that's the reason that the `BasicBlockNodeParams` are configurable: both generation and shrinking on failure are costly, so this is meant to minimize CI testing costs with a conservative default. 
We should definitely run this on larger forests in targeted tests, and this parameter allows us to do so.